Valigator.Newtonsoft.Json/ValigatorJsonPropertyHandler.cs

```diff
@@ -17,12 +17,84 @@ internal abstract class ValigatorJsonPropertyHandler<TObject>
 		{
-			var handlerType = typeof(ValigatorJsonPropertyHandler<,>).MakeGenericType(typeof(TObject), property.PropertyType.GetGenericArguments()[0]);
+			var propertyHandlerType = property.PropertyType.IsValigatorDataType() ? typeof(ValigatorJsonPropertyHandler<,>) : typeof(NonValigatorDataJsonPropertyHandler<,>);
+
+			var propertyType = property.PropertyType.IsValigatorDataType() ? property.PropertyType.GetGenericArguments()[0] : property.PropertyType;
+
+			var handlerType = propertyHandlerType.MakeGenericType(typeof(TObject), propertyType);
 
 			var createMethod = handlerType.GetMethod(nameof(ValigatorJsonPropertyHandler<TObject, bool>.Create), BindingFlags.Public | BindingFlags.Static, Type.DefaultBinder, new[] { typeof(PropertyInfo) }, null);
 
 			return (ValigatorJsonPropertyHandler<TObject>)createMethod.Invoke(null, new[] { property });
 		}
+
+		internal static Func<TObject, TValue> CreateGetter<TValue>(PropertyInfo property)
+		{
+			var objParameter = Expression.Parameter(typeof(TObject), "obj");
+			var propertyParameter = Expression.Constant(property);
+
+			var getValueMethod = property.GetType().GetMethod(nameof(PropertyInfo.GetValue), new[] { typeof(object) });
+			var getterExpression = Expression.Convert(Expression.Call(propertyParameter, getValueMethod, objParameter), property.PropertyType);
+			var getter = property.CanRead ? Expression.Lambda<Func<TObject, TValue>>(getterExpression, objParameter).Compile() : null;
+
+			return getter;
+		}
+
+		internal static Action<TObject, TValue> CreateSetter<TValue>(PropertyInfo property)
+		{
+			var objParameter = Expression.Parameter(typeof(TObject), "obj");
+			var propertyParameter = Expression.Constant(property);
+
+			var valueParameter = Expression.Parameter(typeof(TValue), "value");
+			var setValueMethod = property.GetType().GetMethod(nameof(PropertyInfo.SetValue), new[] { typeof(object), typeof(object) });
+			var setProperty = Expression.Call(propertyParameter, setValueMethod, objParameter, Expression.Convert(valueParameter, typeof(object)));
+			var setter = property.CanWrite ? Expression.Lambda<Action<TObject, TValue>>(setProperty, objParameter, valueParameter).Compile() : null;
+
+			return setter;
+		}
+	}
+
+	internal class NonValigatorDataJsonPropertyHandler<TObject, TValue> : ValigatorJsonPropertyHandler<TObject>
+	{
+		private readonly Func<TObject, TValue> _getValue;
+		private readonly Action<TObject, TValue> _setValue;
+
+		public override bool CanRead => _getValue != null;
+		public override bool CanWrite => _setValue != null;
+
+		public NonValigatorDataJsonPropertyHandler(Func<TObject, TValue> getValue, Action<TObject, TValue> setValue)
+		{
+			_getValue = getValue;
+			_setValue = setValue;
+		}
+
+		public override void ReadProperty(JsonReader reader, JsonSerializer serializer, TObject obj)
+		{
+			if (_setValue != null)
+				_setValue(obj, serializer.Deserialize<TValue>(reader));
```

Comment:
In the case where it isn't a `Data<>` just use the serializer as usual to deserialize, but we need to set using the _setValue.